analyzers/src/SonarAnalyzer.CSharp/Rules/TestClassShouldHaveTestMethod.cs

```diff
@@ -53,7 +53,7 @@ public sealed class TestClassShouldHaveTestMethod : SonarDiagnosticAnalyzer
             context.RegisterSyntaxNodeActionInNonGenerated(
                 c =>
                 {
-                    if (c.ContainingSymbol.Kind != SymbolKind.NamedType)
+                    if (c.IsRedundantPositionalRecordContext())
```

Comment:
`c =>` can be on previous line. 
With moving the cast before the `if`, we can merge all 3 conditions